### PR TITLE
SOC-530 As a user, I want better registration and account-related emails

### PR DESCRIPTION
**Pull request by jsutterfield**
This PR updates 3 more emails to use the new templates. They are emails received when registering an account to confirm the email, the email which is sent 1 week after registration to remind the user to confirm their email if they haven't done so yet, and finally the confirmation email a user receives when they update their email address.

Ticket: https://wikia-inc.atlassian.net/browse/SOC-530

### CODE REVIEW

File: extensions/wikia/Email/Controller/EmailConfirmationController.class.php [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+			];
+
+			return $form;
+
+	}
+}
+
+class EmailConfirmationController extends AbstractEmailConfirmationController {
+
+	const TRACKING_CATEGORY = "ConfirmationMail";
```
Comment (jsutterfield) [inline extensions/wikia/Email/Controller/EmailConfirmationController.class.php]: Just calling out these `TRACKING_CATEGORY` constants. This is how we'll categorize emails in sendgrid going forward and came out of that P2 (SOC-887).

File: extensions/wikia/Email/Controller/EmailConfirmationController.class.php [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+			];
+
+			return $form;
+
+	}
+}
+
+class EmailConfirmationController extends AbstractEmailConfirmationController {
+
+	const TRACKING_CATEGORY = TrackingCategories::EMAIL_CONFIRMATION;
```
Comment (jsutterfield) [inline extensions/wikia/Email/Controller/EmailConfirmationController.class.php]: Just calling out these TRACKING_CATEGORY constants. This is how we'll categorize emails in sendgrid going forward and came out of that P2 (SOC-887).

File: extensions/wikia/Email/EmailController.class.php [hunk 3]
```diff
@@ -72,12 +75,12 @@ public function init() {
 				$this->getVal( 'replyToName', $noReplyName )
 			);
 
-			$fromAddress = $this->getVal( 'fromAddress', '' );
+			$fromAddress = $this->getVal( 'fromAddress', $this->wg->PasswordSender );
```
Comment (jsutterfield) [inline extensions/wikia/Email/EmailController.class.php]: fromAddress and fromName were originally being passed into the controller b/c the first set of emails we did (the watched article emails) had confusing and custom logic for these values. Those emails will still pass in custom values for these fields, but most other emails default to using `$wgPasswordSender` and `$wgPasswordSenderName` so we can just make those the defaults.

File: extensions/wikia/GlobalWatchlist/GlobalWatchlist.i18n.php [hunk 4]
```diff
@@ -0,0 +1,425 @@
+<?php
+
+$messages = array();
+
+$messages['en'] = array(
+	"tog-watchlistdigest" => "E-mail me a Weekly Watchlist Digest",
+	"tog-watchlistdigestclear" => "Mark all pages from my Weekly Watchlist Digest as visited",
```
Comment (jsutterfield) [inline extensions/wikia/GlobalWatchlist/GlobalWatchlist.i18n.php]: As part of SOC-530 I took out this entire file, but realized that we still need these 2 messages which are part of Special:Preferences for the weekly digest.

File: extensions/wikia/GlobalWatchlist/GlobalWatchlistBot.class.php [hunk 5]
```diff
@@ -114,9 +111,6 @@ public function sendDigestToUser( $userID ) {
 
 		$params = [
 			'targetUser' => User::newFromId( $userID )->getName(),
-			'replyToAddress' => self::REPLY_ADDRESS,
-			'fromAddress' => self::FROM_ADDRESS,
-			'fromName' => self::FROM_NAME,
```
Comment (jsutterfield) [inline extensions/wikia/GlobalWatchlist/GlobalWatchlistBot.class.php]: These are now just using the defaults in the controller which have the same values.

File: extensions/wikia/Email/Controller/EmailConfirmationController.class.php [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+	/**
+	 * @template emailConfirmation
+	 */
+	public function body() {
+		$this->response->setData( [
+			'salutation' => $this->getSalutation(),
+			'summary' => $this->getSummary(),
+			'buttonLink' => $this->confirmUrl,
+			'buttonText' => $this->getMessage( 'emailext-emailconfirmation-button-text' ),
```
Comment (garthwebb) [inline extensions/wikia/Email/Controller/EmailConfirmationController.class.php]: I think this needs a ```->text``` or ```->escaped``` after it.

File: includes/User.php [hunk 7]
```diff
@@ -3661,23 +3666,56 @@ public function sendConfirmationMail( $type = 'created', $mailtype = "Confirmati
...
+		return $mailType == "ReConfirmationMail";
+	}
+
+	private function sendUsingEmailExtension( $emailController, $url ) {
+		$params = [
+			'targetUser' => $this->getName(),
+			'confirmUrl' => $url,
+		];
+
+		$responseData =  F::app()->sendRequest( $emailController, 'handle', $params )->getData();
```
Comment (garthwebb) [inline includes/User.php]: Extra space between ```=``` and ```F::app```

File: includes/User.php [hunk 8]
```diff
@@ -3661,23 +3666,56 @@ public function sendConfirmationMail( $type = 'created', $mailtype = "Confirmati
...
+			return Status::newFatal( $responseData['error'] );
+		}
+
+	}
+
 	/**
-	 * Confirmation after change the emial
+	 * Confirmation after change the email
 	 *
 	 * @return \types{\bool,\type{WikiError}} True on success, a WikiError object on failure.
```
Comment (garthwebb) [inline includes/User.php]: Maybe we can update this to the simpler ```WikiError|bool```

File: extensions/wikia/Email/i18n/EmailConfirmation.i18n.php [hunk 9]
```diff
@@ -0,0 +1,211 @@
+<?php
+$messages = array();
+
+$messages['en'] = array(
+	'emailext-emailconfirmation-subject' => 'Confirm your email and get started on Wikia!',
+	'emailext-emailconfirmation-summary' => 'You’re SO CLOSE! Click the link below to confirm your email address and get started on Wikia:',
+	'emailext-emailconfirmation-footer-1' => 'Why confirm your email, you ask? You’ll get the latest updates on any of topics you follow, messages from your fellow fans, and lots more delivered to the comfort of your inbox.  You can also create your very own wikias and get other special permissions exclusive to fully confirmed members.',
```
Comment (mklucsarits) [inline extensions/wikia/Email/i18n/EmailConfirmation.i18n.php]: "on any of topics" doesn't sound right. Is there a word missing there?

File: extensions/wikia/Email/templates/avatarLayout.mustache [hunk 10]
```diff
@@ -30,20 +30,8 @@
...
-		<tr>
-			<td colspan="2">
-				<div class="content-footer-messages">
-					{{#contentFooterMessages}}
-						<p>{{{.}}}</p>
-					{{/contentFooterMessages}}
-				</div>
-			</td>
-		</tr>
+		{{> contentFooterMessages}}
```
Comment (mklucsarits) [inline extensions/wikia/Email/templates/avatarLayout.mustache]: :+1:

File: extensions/wikia/Email/i18n/EmailConfirmation.i18n.php (hunk 9 above)
Comment (jsutterfield) [inline extensions/wikia/Email/i18n/EmailConfirmation.i18n.php]: Good catch. I think it needs a the. I'll add it.

**Pull request closed without merging**